### DISCUSSION TITLE
create a base "library" which could be used by other frameworks like Nativescript

### DISCUSSION THREAD

**Discussion by farfromrefug** [getsentry/sentry-react-native#3850] [Ideas]
I am the maintainer of https://github.com/nativescript-community/sentry  which is a sentry plugin for Nativescript.
That plugins shared almost 100% of the same typescript code as the react native one. Only the wrappers and some platform specific things (like the route navigation tracing) are different.
I have been maintaining it for years now and it is quite some work as i have to follow every commit and transpose everything.

I think it would be fairly easy to have a base library (typescript one) which could be used by the react native and the nativescript plugin. It could also lead the way for other platforms.
I am  obviously ok working on this but i cant do that without the sentry team. All the design / architecture would have to come from the sentry team.

What do you think about the idea?

**Comment by krystofwoldrich** [comment]
Hi, 
excuse the long response time.

First of all, thank you for maintaining https://github.com/nativescript-community/sentry, that's awesome.

I agree we should duplicate code across JS based SDKs, for that purpose `@sentry/core` exists (and some other like `@sentry/utils`...). 

Could you share some specific example of the native script and RN SDK use the same but each has their own copy of the code?

Maybe some of these functions/block could be moved to the main https://github.com/getsentry/sentry-javascript repository.

**Reply by farfromrefug** [reply]
@krystofwoldrich thanks for answering!
On this repo (sentry-react-native) almost all the typescript code is shared. Differences:
* `wrapper.ts` is specific and we use our own implementation. So not shared
*  all the react / native part is not shared. Obviously we dont want to share it
* consequently touchevents / tracing / hermes profiling (for now we plan on supporting hermes too) / native error handler (we use our own) / expo context /spotlight are not to be shared.
* version is also obviously not shared.

I think that s about it. So it would great to have kind of all the rest moved to shared "mobile-core" or something (not sure it should even work for desktops like tauri/electron). Maybe have base classes for integrations like tracing, native error handler... so that each platform can easily implement their own.

Having that would make the task of maintaining another platform than react-native a 1000 times easier.

Thanks again for taking the time to answer me
